libbeat/esleg/eslegclient/connection.go

```diff
@@ -284,6 +287,9 @@ func (conn *Connection) Connect() error {
 	if conn.log == nil {
 		conn.log = logp.NewLogger("esclientleg")
 	}
+
+	conn.reqsContext, conn.cancelReqs = context.WithCancel(context.Background())
```

Comment:
This replaces the context created in `NewConnection` without cancelling it first.

Reply:
Indeed it does, however Connection is not goroutine safe, so if there are in-flight requests, this line of code should never execute. Technically this should only replace a context that has already been cancelled or is not being used anymore. Anyways, I can cancel the previous context just in case.